testing/integration-tests/src/light_client/mod.rs

```diff
@@ -93,42 +126,61 @@ async fn storage_plain_lookup(api: &Client) -> Result<(), subxt::Error> {
         .fetch_or_default(&addr)
         .await?;
 
+    println!("Storage lookup took {:?}\n", now.elapsed());
+
     assert!(entry > 0);
 
     Ok(())
 }
 
 // Make a dynamic constant query for `System::BlockLenght`.
 async fn dynamic_constant_query(api: &Client) -> Result<(), subxt::Error> {
+    let now = std::time::Instant::now();
+    println!("In dynamic_constant_query");
+
     let constant_query = subxt::dynamic::constant("System", "BlockLength");
     let _value = api.constants().at(&constant_query)?;
 
+    println!("Dynamic constant query took {:?}\n", now.elapsed());
+
     Ok(())
 }
 
 // Fetch a few all events from the latest block and decode them dynamically.
 async fn dynamic_events(api: &Client) -> Result<(), subxt::Error> {
+    let now = std::time::Instant::now();
+    println!("In dynamic_events");
+
     let events = api.events().at_latest().await?;
 
     for event in events.iter() {
         let _event = event?;
+
+        println!("Event decoding took {:?}", now.elapsed());
     }
 
+    println!("Dynamic events took {:?}\n", now.elapsed());
+
     Ok(())
 }
 
 #[tokio::test]
 async fn light_client_testing() -> Result<(), subxt::Error> {
+    tracing_subscriber::fmt::init();
+
+    let now = std::time::Instant::now();
     let api: LightClient<PolkadotConfig> = LightClientBuilder::new()
         .build_from_url("wss://rpc.polkadot.io:443")
         .await?;
 
-    non_finalized_headers_subscription(&api).await?;
+    println!("Light client initialization took {:?}\n", now.elapsed());
+
+    // non_finalized_headers_subscription(&api).await?;
     finalized_headers_subscription(&api).await?;
-    runtime_api_call(&api).await?;
-    storage_plain_lookup(&api).await?;
-    dynamic_constant_query(&api).await?;
-    dynamic_events(&api).await?;
+    // runtime_api_call(&api).await?;
+    // storage_plain_lookup(&api).await?;
+    // dynamic_constant_query(&api).await?;
+    // dynamic_events(&api).await?;
```

Comment:
Should these still be commented out?

Reply:
Yep, I've added a few debug logs in https://github.com/paritytech/subxt/pull/1400/commits/f14bf9a47457de79475eb683962a0c8d3f6c3681.
This is to help reproduce the issue from smoldot faster https://github.com/smol-dot/smoldot/issues/1638. 

I suggest putting this PR on pending until another smoldot version is published on crates io :D

Reply:
Sounds like a plan :)